Stub MQ docs (#10347)
galaxyproject/galaxy: doc/source/admin/mq.md

# Message queue

Galaxy uses a message queue internally for communicating between processes. 
This is powered by the [Kombu MQ library](https://docs.celeryproject.org/projects/kombu/en/stable/). 
For example, when reloading the toolbox or locking job execution,
the process that handled that particular request will tell all others to also reload, lock jobs, etc. 

<!-- TODO Discuss Galaxies benefit of using an MQ -->

<!-- TODO the types of messages being passed, more detailed than above -->

This is configured via the [amqp_internal_connection](https://docs.galaxyproject.org/en/latest/admin/config.html#amqp-internal-connection)

> option in the galaxy.yaml. For connection examples, see the [kombu connection documentation](http://docs.celeryproject.org/projects/kombu/en/latest/userguide/connections.html).

Reviewer: ```suggestion
option in `galaxy.yml`. For connection examples, see the [Kombu connection documentation](http://https://docs.celeryproject.org/projects/kombu/en/stable/userguide/connections.html).
```

> See the [url specification](https://docs.celeryproject.org/projects/kombu/en/stable/userguide/connections.html#urls) on that page for more information on configuring different transports.

Reviewer: ```suggestion
See the [URL specification](https://docs.celeryproject.org/projects/kombu/en/stable/userguide/connections.html#urls) on that page for more information on configuring different transports.
```

<!-- TODO copy in an actual rabbit MQ example from the link above 
```
amqp_internal_connection: 
```
-->

> Unconfigured, galaxy will first attempt to use

Reviewer: ```suggestion
By default, Galaxy will first attempt to use
```

your specified [database_connection](https://docs.galaxyproject.org/en/latest/admin/config.html#database-connection).  

> If that's not specified either, Galaxy will automatically create and use a separate sqlite

Reviewer: ```suggestion
If that's not specified either, Galaxy will automatically create and use a separate SQLite
```

> database located in your <galaxy>/database folder (indicated in the commented out line below).

Reviewer: ```suggestion
database located in your <galaxy>/database folder.
```

## Transports
Kombu and Galaxy support a variety of MQ transport/server options. 

> [RabbitMQ](https://www.rabbitmq.com/) via AMQP is the most popular.

Reviewer: ```suggestion
[RabbitMQ](https://www.rabbitmq.com/) via AMQP is the most popular for production deployments.
```
